### PR TITLE
[No Ticket] Full MockOptimizely Removal PR Fixes/Cleanup

### PR DESCRIPTION
**Pull request by scottkicks**
# 📲 What

Regarding my [Full MockOptimizely Removal PR](https://github.com/kickstarter/ios-oss/pull/1823)
There are some further changes and cleanup that can be done. 

# 🤔 Why

Its important to be thorough in code cleanup, making sure to really understand why certain changes are being made and how best to address removing and/or cleaning up so that we can guarantee that our app is better for it.

# 🛠 How

- Adding tests around our `.ksr_remoteConfigClientConfigured` & `.ksr_remoteConfigClientConfigurationFailed` notifications
- Simplifying properties like `paymentSheetEnabled` & `campaignTabEnabled` that were previously updated
- Removing redundant tests
- Removing Snapshots associated with snapshot tests that were removed and making sure only affected snapshots are updated

# ✅ Acceptance criteria

- [ ] all tests pass

# ⏰ TODO

- [x] Discuss snapshot changes from old PR with Mubarak to get better aligned on why they were so many changes in the first place and how best to approach next steps.

### CODE REVIEW

File: Kickstarter-iOS/Features/Comments/Controllers/CommentsViewControllerTests.swift [hunk 1]
```diff
@@ -174,35 +174,6 @@ internal final class CommentsViewControllerTests: TestCase {
     }
   }
 
-  func testView_CurrentUser_LoggedIn_IsBacking() {
```
Comment (msadoon) [inline Kickstarter-iOS/Features/Comments/Controllers/CommentsViewControllerTests.swift:177]: Always always always remember to remove screenshots if a test is removed. :)

**Review by msadoon — CHANGES_REQUESTED** [review]
So close to perfect, haha, just remember if we remove a test in a view controller to remove the corresponding screenshots.

File: Kickstarter-iOS/Features/Comments/Controllers/CommentsViewControllerTests.swift (hunk 1 above)
Comment (scottkicks) [reply Kickstarter-iOS/Features/Comments/Controllers/CommentsViewControllerTests.swift:177]: sorry. those are gone now.

**Review by msadoon — APPROVED** [review]
Update the milestone to 5.9.0 - but don't merge this. Let's merge the marketing version update for 5.8.0 first, close out milestone 5.8.0, update the release branch, then merge this in.

**Pull request merged**